contracts/src/NativeTokenDestination.sol

```diff
@@ -348,7 +375,7 @@ contract NativeTokenDestination is
     /**
      * @dev Mints coins to the recipient through the NativeMinter precompile.
      */
-    function _mint(address recipient, uint256 amount) private {
+    function _mintNativeCoin(address recipient, uint256 amount) private {
```

Comment:
Just now realizing that the native minter precompile's method is `mintNativeCoin` and not `mintNativeToken`. Do you know if there's any technical distinction between the two terms in this context? If they're identical, it's a bit unfortunate that our bridge contracts use "Token" and the precompile uses "Coin."

Reply:
Technically, "coin" refers to an asset on its own independent blockchain, and "tokens" are assets created on top of blockchains (other than that blockchain's coin presumably). I do think we've slightly misused them here, but I'm also not sure those definitions fit nicely in this cross-chain world. For instance, does an ERC20 token become a coin if bridged to another chain where it's represented as that chain's native asset? 

[ref](https://www.coinbase.com/learn/crypto-basics/what-is-the-difference-between-a-coin-and-a-token)